board/common/overlay/etc/init.d/base

```diff
@@ -24,3 +24,37 @@ msg_background() {
     test -n "$1" && echo $1 || echo "pending"
 }
 
+prepare_conf() {
+    # $1 - actual config file
+    # $2 - system-provided config file
+    # $3 - user-provided config file
+    
+    # long story short:
+    #  * user conf file takes precedence, if present
+    #  * system conf file is used by default, if actual file absent
+    
+    actual_conf="$1"
+    system_conf="$2"
+    user_conf="$3"
+    
+    if [[ -n "$user_conf" && -e "$user_conf" ]]; then
+        cp -rf "$user_conf" "$actual_conf"
+        grep -E "/boot .*ro[\s,]" /proc/mounts &>/dev/null
+        RO=$?
+        test $RO == 0 && mount -o remount,rw /boot
+        rm -rf $user_conf
+        test $RO == 0 && mount -o remount,ro /boot
+    fi
+
+    if [[ ! -e "$actual_conf" && -e "$system_conf" ]]; then
+        cp "$system_conf" "$actual_conf"
+    fi
+
+    # we want only Unix newlines in conf files
```

Comment:
@ccrisan I would move this part into the above `if [[ -n "$user_conf" && -e "$user_conf" ]]; then` block, just after `cp -rf "$user_conf" "$actual_conf"`, to minimize the amount of processing required on every start up. We only need to do the newline conversion if the file coming from the user. Our system default config files should already have unix newline.

Reply:
Very good point. Fixed via 92bd4b031d8ac63928939290c852df64311fc518.